core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -390,16 +390,18 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         case Some(executorInfo) =>
           // This must be synchronized because variables mutated
           // in this block are read when requesting executors
-          val killed = CoarseGrainedSchedulerBackend.this.synchronized {
+          val lossReason = CoarseGrainedSchedulerBackend.this.synchronized {
             addressToExecutorId -= executorInfo.executorAddress
             executorDataMap -= executorId
             executorsPendingLossReason -= executorId
-            executorsPendingDecommission -= executorId
-            executorsPendingToRemove.remove(executorId).getOrElse(false)
+            val decommissioned = executorsPendingDecommission.remove(executorId)
+            executorsPendingToRemove
+              .remove(executorId).filter(killedByDriver => killedByDriver).map(_ => ExecutorKilled)
```

Comment:
This scala statement is quite a mouthful ? Would an if condition make it simpler ?

Reply:
Yeah..looks better. (I actually tried both ways before and was hesitant between them...but I tried the scala style at the end..)